javascript/javascript2/week2/hyf-bay/main.js

```diff
@@ -0,0 +1,62 @@
+console.log('Script loaded');
+
+const products = getAvailableProducts();
+const productsUl = document.querySelector('section.products ul');
+console.log(productsUl);
+
+function renderProducts(products) {
+    productsUl.innerHTML=''
+    products.forEach(product => {
+        const li = document.createElement('li');
+
+        let shipsToHTML = '';
+        product.shipsTo.forEach(country => shipsToHTML += `<li>${country}</li>`);
+
+        li.innerHTML = `
+            <ul>
+                <li>${product.name}</li>
+                <li>${product.price}</li>
+                <li>${product.rating}</li>
+                <ul class="ships-to">${shipsToHTML}</ul>
+            </ul>
+        `;
+        productsUl.appendChild(li);
+    });
+}
+
+renderProducts(products);
+
+const userInput = document.getElementById("inputName");
+const userOption = document.getElementById("selectOption");
+const userCountry = document.getElementById("countryOption");
+const userPriceRange = document.getElementById("start");
+
+userInput.addEventListener("keyup", ()=>{
+    const searchedProduct = document.getElementById("inputName").value.toLowerCase()
+    return searchedProduct ? renderProducts(products.filter(a=>a.name.toLowerCase().includes(searchedProduct))) : 
+    renderProducts(products)   
+})
+userOption.addEventListener("change", ()=>{
+    const searchedOption = document.getElementById("selectOption").value
+    return searchedOption=="cheap" ? renderProducts(products.sort((a,b)=>a.price-b.price)) : 
+    (searchedOption=="expensive" ? renderProducts(products.sort((a,b)=>b.price-a.price)) : 
+    renderProducts(products.sort((a,b)=>a.name>b.name && 1 || -1)))  
+})
```

Comment:
You should avoid chaining ternary operators ("? :").
It makes code hard to read. 
```js
userOption.addEventListener("change", (event) => {
    const searchedOption = event.target.value
    if (searchedOption === "cheap") {
        return renderProducts(products.sort((a,b)=>a.price-b.price))
    } else if (searchedOption === "expensive") {
        return renderProducts(products.sort((a,b)=>b.price-a.price))
    } else {
        return renderProducts(products.sort((a,b)=>a.name>b.name && 1 || -1))
    }
})
```

Reply:
haha... thanks ... i always do this... everytime when i do like this..they give feedback to use ternary...